core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/PillarSwapMoveSelectorFactory.java

```diff
@@ -26,17 +26,16 @@ protected MoveSelector<Solution_> buildBaseMoveSelector(HeuristicConfigPolicy<So
             SelectionCacheType minimumCacheType, boolean randomSelection) {
         PillarSelectorConfig leftPillarSelectorConfig =
                 Objects.requireNonNullElseGet(config.getPillarSelectorConfig(), PillarSelectorConfig::new);
-        PillarSelector<Solution_> leftPillarSelector =
-                buildPillarSelector(leftPillarSelectorConfig, configPolicy, minimumCacheType, randomSelection);
         PillarSelectorConfig rightPillarSelectorConfig =
                 Objects.requireNonNullElse(config.getSecondaryPillarSelectorConfig(), leftPillarSelectorConfig);
+        PillarSelector<Solution_> leftPillarSelector =
+                buildPillarSelector(leftPillarSelectorConfig, configPolicy, minimumCacheType, randomSelection);
```

Comment:
Seems like a change for change's sake. I can guess that the idea was to improve the locality of `leftPillarSelector`, but that in turns worsens the locality of `leftPillarSelectorConfig`. 

Six of one, half a dozen of other; I'd revert the change.

Reply:
Without any other context, I'd agree. But this is not about the locality of the variables. It's about the procedure.

The intention is to first focus on the config resolution and then on building the selectors.

The reason for keeping config resolution together is that it makes it easier to notice and understand that the left pillar config falls back to empty config but the right pillar config _does not_. It falls back to the left pillar config. This is significant and it's important to notice the intention. I think I should add a comment about that.

Other selector factories follow the same structure.